### PR TITLE
Holes and lava do not go out of bounds in  plotter

### PR DESCRIPTION
**Pull request by jaudick**

### CODE REVIEW

File: machine_common_sense/plotter.py [hunk 1]
```diff
@@ -180,10 +180,14 @@ def _initialize_plot(self, scene_config: SceneConfiguration) -> np.ndarray:
 
         img = self._draw_grid_lines(img, buffer)
         img = self._draw_holes(
-            img, scene_config.holes)
+            img, scene_config.holes,
+            scene_config.room_dimensions.x,
+            scene_config.room_dimensions.z)
```
Comment (deanwetherby) [inline machine_common_sense/plotter.py:185]: You could just pass in room_dimensions and let the internal function access the attributes.

File: machine_common_sense/plotter.py [hunk 2]
```diff
@@ -296,10 +300,59 @@ def _draw_room_border(self, img: np.ndarray,
         img[rr, cc] = self.BORDER_COLOR
         return img
 
+    def _plotter_hole_texture_size_reducer(
```
Comment (deanwetherby) [inline machine_common_sense/plotter.py:303]: Function could be broken down into smaller, individually testable chunks which would also help with complexity.

**Comment by deanwetherby** [comment]
Unit tests are great.

**Review by deanwetherby — APPROVED** [review]

**Review by kdrumm — APPROVED** [review]

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]
Good use of type hints and unit tests.

**Pull request merged**